### PR TITLE
映射文档 No. 26

### PR DESCRIPTION
**Pull request by zeyuxionghust**
完成分组No.26文档映射，其中新提交9个md文件。
torch.sparse.sum因Paddle功能缺失，未添加。

### CODE REVIEW

File: docs/guides/model_convert/convert_from_pytorch/api_difference/fft/torch.fft.fftshift.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+## [ 仅 paddle 参数更多 ] torch.fft.fftshift
```
Comment (Fancy-hjyp) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/fft/torch.fft.fftshift.md]: 这里我们可以忽略name参数，所以分类写 `仅参数名不一致` 即可，参数映射部分也可以直接忽略name~

File: docs/guides/model_convert/convert_from_pytorch/api_difference/fft/torch.fft.fftshift.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+torch.fft.fftshift(input, dim=None)
+```
+
+### [paddle.fft.fftshift](https://www.paddlepaddle.org.cn/documentation/docs/zh/api/paddle/fft/fftshift_cn.html)
+
+```python
+paddle.fft.fftshift(x, axes=None, name=None)
+```
+
+Paddle 相比 PyTorch 支持更多其他参数，具体如下：
```
Comment (Fancy-hjyp) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/fft/torch.fft.fftshift.md]: 这句话也可以相应修改一下

File: docs/guides/model_convert/convert_from_pytorch/api_difference/fft/torch.fft.fftshift.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+### 参数映射
+| PyTorch       | PaddlePaddle | 备注                                                   |
+| ------------- | ------------ | ------------------------------------------------------ |
+| input           | x           | 输入 Tensor，仅参数名不一致。               |
+| dim           | axes           | 进行移动的轴，仅参数名不一致。               |
+| -           | name           | 网络前缀标识，PyTorch 无此参数，Paddle 保持默认即可。               |
```
Comment (Fancy-hjyp) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/fft/torch.fft.fftshift.md]: 可以直接删去name参数

File: docs/guides/model_convert/convert_from_pytorch/api_difference/init/torch.nn.init.calculate_gain.md [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+torch.nn.init.calculate_gain(nonlinearity, param=None)
+```
+
+### [paddle.nn.initializer.calculate_gain](https://www.paddlepaddle.org.cn/documentation/docs/zh/api/paddle/nn/initializer/calculate_gain_cn.html)
+
+```python
+paddle.nn.initializer.calculate_gain(nonlinearity, param=None)
+```
+
+两者参数和用法完全一致
```
Comment (Fancy-hjyp) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/init/torch.nn.init.calculate_gain.md]: 完全一致的辛苦也写一下参数映射部分~

File: docs/guides/model_convert/convert_from_pytorch/api_difference/others/torch.random.manual_seed.md [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+torch.random.manual_seed(seed)
+```
+
+### [paddle.seed](https://www.paddlepaddle.org.cn/documentation/docs/zh/api/paddle/seed_cn.html)
+
+```python
+paddle.seed(seed)
+```
+
+两者参数和用法完全一致
```
Comment (Fancy-hjyp) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/others/torch.random.manual_seed.md]: 完全一致的辛苦也写一下参数映射部分~

File: docs/guides/model_convert/convert_from_pytorch/api_difference/sparse/torch.sparse.addmm.md [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+```python
+paddle.sparse.addmm(input, x, y, beta=1.0, alpha=1.0, name=None)
+```
+
+两者功能一致，仅参数名不一致，具体如下：
+
+### 参数映射
+
+ PyTorch |  PaddlePaddle |  备注
```
Comment (Fancy-hjyp) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/sparse/torch.sparse.addmm.md]: 这里的表格格式是不是不太对，每行前面没有 `| `

File: docs/guides/model_convert/convert_from_pytorch/api_difference/sparse/torch.sparse.mm.md [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+
+```python
+paddle.sparse.matmul(x, y, name=None)
+```
+
+两者功能一致，仅参数名不一致，具体如下：
+
+### 参数映射
+
+ PyTorch |  PaddlePaddle |  备注
```
Comment (Fancy-hjyp) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/sparse/torch.sparse.mm.md]: 表格格式调整一下

File: docs/guides/model_convert/convert_from_pytorch/api_difference/sparse/torch.sparse.softmax.md [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+
+Pytorch 相比 Paddle 支持更多其他参数，具体如下：
+
+### 参数映射
+
+ PyTorch  | PaddlePaddle |  备注
+ --------|  -------------| --------------------------------------------------------------------------------------
+ input  |x       |  输入的 Tensor，仅参数名不一致。
+ dim   |      axis|   输入的第二个 Tensor，仅参数名不一致。
+ dtype | -  | 指定数据类型，PaddlePaddle 无此功能。
```
Comment (Fancy-hjyp) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/sparse/torch.sparse.softmax.md]: 这种需要添加转写示例，通过astype()函数显式指定类型

Comment (zeyuxionghust) [reply docs/guides/model_convert/convert_from_pytorch/api_difference/sparse/torch.sparse.softmax.md]: 可以给我个demo吗

Comment (zeyuxionghust) [reply docs/guides/model_convert/convert_from_pytorch/api_difference/sparse/torch.sparse.softmax.md]: 感谢指出问题。
这个sparse的Tensor可能无法直接用astype，故使用paddle.sparse.cast转化。
有时间麻烦再check一下

File: docs/guides/model_convert/convert_from_pytorch/api_difference/sparse/torch.sparse.softmax.md [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+
+Pytorch 相比 Paddle 支持更多其他参数，具体如下：
+
+### 参数映射
+
+ |PyTorch  | PaddlePaddle |  备注       |
+ |--------|  -------------| --------------------------------------------------------------------------------------|
+ |input  |x       |  输入的稀疏 Tensor，仅参数名不一致。|
+ |dim   |      axis|   指定对输入 SparseTensor 计算 softmax 的轴，Paddle 的默认值：-1。仅参数名不一致。|
+ |dtype | -  | 指定数据类型，可选项，Pytorch 默认值为 None，PaddlePaddle 无此功能。|
```
Comment (Fancy-hjyp) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/sparse/torch.sparse.softmax.md]: 备注可以加一句，`需要进行转写`

File: docs/guides/model_convert/convert_from_pytorch/api_difference/sparse/torch.sparse.softmax.md [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+### 转写示例
+#### dytpe：指定数据类型
+```Python
+# torch
```
Comment (Fancy-hjyp) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/sparse/torch.sparse.softmax.md]: 注释写成`PyTorch写法`， `Paddle写法` 这样的吧，保持统一~

File: docs/guides/model_convert/convert_from_pytorch/api_difference/sparse/torch.sparse.softmax.md [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+torch.sparse.softmax(x,-1,dtype=torch.float32)
+
+# paddle 转化 values 的 dtype 到 float32 数据类型
+x = paddle.sparse.cast(x, index_dtype=None, value_dtype='float32')
+paddle.sparse.nn.functional.softmax(x,-1)
```
Comment (Fancy-hjyp) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/sparse/torch.sparse.softmax.md]: 觉得先计算softmax，再修改type更好一些，因为用户可能不想修改x的type
```python
y = paddle.sparse.nn.functional.softmax(x)
y = paddle.sparse.cast(y, value_dtype='float32')
```

File: docs/guides/model_convert/convert_from_pytorch/api_difference/sparse/torch.sparse.softmax.md [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+ |input  |x       |  输入的稀疏 Tensor，仅参数名不一致。|
+ |dim   |      axis|   指定对输入 SparseTensor 计算 softmax 的轴，Paddle 的默认值：-1。仅参数名不一致。|
+ |dtype | -  | 指定数据类型，可选项，Pytorch 默认值为 None，PaddlePaddle 无此功能。|
+### 转写示例
+#### dytpe：指定数据类型
+```Python
+# torch
+torch.sparse.softmax(x,-1,dtype=torch.float32)
+
+# paddle 转化 values 的 dtype 到 float32 数据类型
```
Comment (Fancy-hjyp) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/sparse/torch.sparse.softmax.md]: 直接写 Paddle写法 即可~

File: docs/guides/model_convert/convert_from_pytorch/api_difference/sparse/torch.sparse.softmax.md (hunk 11 above)
Comment (zeyuxionghust) [reply docs/guides/model_convert/convert_from_pytorch/api_difference/sparse/torch.sparse.softmax.md]: 可是Pytorch API是这样写的，是before operation，很多dtype都是先转换
dtype– the desired data type of returned tensor. If specified, the input tensor is casted to dtype before the operation is performed. This is useful for preventing data type overflows. Default: None

Comment (zeyuxionghust) [reply docs/guides/model_convert/convert_from_pytorch/api_difference/sparse/torch.sparse.softmax.md]: 参考torch.sparse.softmax和torch.nn.functional.softmax的dtype参数
[torch.sparse.softmax](https://pytorch.org/docs/1.13/generated/torch.sparse.softmax.html#torch.sparse.softmax)
[torch.nn.functional.softmax](https://pytorch.org/docs/1.13/generated/torch.nn.functional.softmax.html?highlight=softmax#torch.nn.functional.softmax)

Comment (zeyuxionghust) [reply docs/guides/model_convert/convert_from_pytorch/api_difference/sparse/torch.sparse.softmax.md]: > 因为用户可能不想修改x的type

明白。已修改为y=paddle.sparse.cast(x, value_dtype='float32')，避免x的dtype被修改。
但还是保持了先转换，符合API文档和源码的流程。

**Review by Fancy-hjyp — APPROVED** [review]
LGTM

**Comment by zhwesky2010** [comment]
@Tomoko-hjf 功能缺失的需要根据模板要求：在最外面的映射表里进行标注即可，不需要写里面的api_difference文档

**Review by zhwesky2010 — APPROVED** [review]

**Comment by Fancy-hjyp** [comment]
> @Tomoko-hjf 功能缺失的需要根据模板要求：在最外面的映射表里进行标注即可，不需要写里面的api_difference文档

好的

**Comment by luotao1** [comment]
hi, @zeyuxionghust 
- 非常感谢你对飞桨框架的贡献，我们正在运营一个PFCC组织，会通过定期分享技术知识与发布开发者主导任务的形式持续为飞桨框架做贡献，详情可见 https://github.com/luotao1 主页说明。
- 如果你对PFCC有兴趣，请发送邮件至 [ext_paddle_oss@baidu.com](mailto:ext_paddle_oss@baidu.com)，我们会邀请你加入~
